### ISSUE TITLE
[Bug]: Download Failed

### ISSUE THREAD

**Issue by ghostboy70144** [th-ch/youtube-music#2628]
### Preflight Checklist

- [X] I use the latest version of YouTube Music (Application).
- [X] I have searched the [issue tracker](https://github.com/th-ch/youtube-music/issues) for a bug report that matches the one I want to file, without success.
- [X] I understand that **th-ch/youtube-music has NO affiliation with Google or YouTube**

### YouTube Music (Application) Version

3.6.2

### Checklists

- [ ] I use the portable version of the YouTube Music Application.
- [x] I can reproduce this issue in the [official version of (WEB) YTM](https://music.youtube.com).

### What operating system are you using?

Windows

### Operating System Version

11

### What CPU architecture are you using?

x64

### Reproduction steps

![image](https://github.com/user-attachments/assets/db139211-80d7-447c-9514-15c5d7cf04e8)

Whenver, i try to download a song from this application it give me this error. This did not usually happen today was the first time it occured. I have also redownloaded the latest version 3.6.2 and its still not working.

### Expected Behavior

The music i wanted to downlaod would have easily been downloaded.

### Actual Behavior

When click download on a music video it usually downloads it and saves it into the set folder.

### Enabled plugins

1.Ad-block
2.Bypass age restrictions
3.Discord Rich Presence
4.Downloader
5.In-App Menu
6.Navigation

**Comment by JellyBrick** [comment]
Fixed by #2681

**Comment by ArvinXO** [comment]
Same issue, using linux mint x64
Disabled & restarted + re-enabled, 1:71 syntax
Running latest version

**Comment by KParthSingh** [comment]
When this is going to be released?

**Comment by JellyBrick** [comment]
@KParthSingh It will be released in 7 days.

**Comment by KParthSingh** [comment]
Alright. thanks

**Comment by Mendungo** [comment]
> Fixed by #2681
We don't know how to fix it that way. I've been using your mod for quite some time and there always seems to be a downloading issue very few updates, i decided not to update the App anymore but it still gets to me somehow on older versions. 
friendly opinion, way too many updates being released often on a pretty stable software which causes issues in the way on things that weren't broken

**Comment by Adrriii** [comment]
Is this actually resolved ? On the current master branch with v12.2.0 of youtubei, the same error pops up when downloading

```Downloading Dakku - Hysteria [34RyDbhWOWA
Error: The server responded with a non 2xx status code
in d509c72e4a5929a4f5be7c7068f15cb19a2da47ff18d506523c9c4273b04c10c
Trace: InnertubeError: The server responded with a non 2xx status code
at C:\Code\youtube-music\node_modules\.pnpm\youtubei.js@12.2.0\node_modules\youtubei.js\dist\src\utils\FormatUtils.js:65:31 {
    info: {
        error_type: 'FETCH_FAILED',
        response: Response {
            status: 403,
            statusText: '',
            headers: Headers {
                'access-control-allow-credentials': 'true',
                'access-control-allow-origin': 'https://www.youtube.com',
                'access-control-expose-headers': 'Client-Protocol, Content-Length, Content-Type, X-Bandwidth-Est, X-Bandwidth-Est2, X-Bandwidth-Est3, X-Bandwidth-App-Limited, X-Bandwidth-Est-App-Limited, X-Bandwidth-Est-Comp, X-Bandwidth-Avg, X-Head-Time-Millis, X-Head-Time-Sec, X-Head-Seqnum, X-Response-Itag, X-Restrict-Formats-Hint, X-Sequence-Num, X-Segment-Lmt, X-Walltime-Ms',
                'content-length': '0',
                'content-type': 'text/plain',
                'cross-origin-resource-policy': 'cross-origin',
                date: 'Mon, 23 Dec 2024 16:23:33 GMT',
                'last-modified': 'Wed, 02 May 2007 10:26:10 GMT',
                server: 'gvs 1.0',
                'timing-allow-origin': 'https://www.youtube.com',
                vary: 'Origin',
                'x-content-type-options': 'nosniff'
            },
            body: ReadableStream { locked: false, state: 'readable', supportsBYOB: false },
            bodyUsed: false,
            ok: false,
            redirected: false,
            type: 'default',
            url: ''
        }
    },
    date: 2024-12-23T16:23:33.963Z,
    version: '12.2.0'
}
at sendError (C:\Code\youtube-music\src\plugins\downloader\main\index.ts:81:11)
at C:\Code\youtube-music\src\plugins\downloader\main\index.ts:484:7
Done: C:\Users\Adri\Music\YouTube Music\Dakku - Hysteria.webm```
